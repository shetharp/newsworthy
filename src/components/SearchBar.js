import React, { Component } from 'react';

class SearchBar extends Component {
  state = { query: '' };

  onFormSubmit = event => {
    event.preventDefault();
    this.props.onSubmit(this.state.query);
  }

  render() {
    return (
      <div>
        <form onSubmit={this.onFormSubmit}>
          <label htmlFor="news-search">News Search</label>
          <input
            type="text"
            name="news-search"
            value={this.state.query}
            onChange={e => this.setState({ query: e.target.value })}
          />
        </form>
      </div>
    );
  }
}

export default SearchBar;
