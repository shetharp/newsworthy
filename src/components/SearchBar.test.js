import React from 'react';
import { shallow, mount, render } from 'enzyme';
import SearchBar from './SearchBar';
// https://devhints.io/enzyme


describe('<SearchBar>', () => {
  it('should render correctly', () => {
    const component = shallow(<SearchBar />);
    expect(component).toMatchSnapshot();
  });

  it('should update state when typing into input', () => {
    const component = mount(<SearchBar />);

    component.find('input')
      .simulate('change', { target: { value: 'climate' } });
    
    expect(component.state().query).toEqual('climate');
    expect(component).toMatchSnapshot();
    component.unmount();
  });

});