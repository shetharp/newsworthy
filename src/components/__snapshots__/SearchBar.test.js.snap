// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar> should maintain state when typing 1`] = `
<SearchBar>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="news-search"
      >
        News Search
      </label>
      <input
        name="news-search"
        onChange={[Function]}
        type="text"
        value="climate"
      />
    </form>
  </div>
</SearchBar>
`;

exports[`<SearchBar> should render correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="news-search"
    >
      News Search
    </label>
    <input
      name="news-search"
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</div>
`;

exports[`<SearchBar> should update state when typing into input 1`] = `
<SearchBar>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="news-search"
      >
        News Search
      </label>
      <input
        name="news-search"
        onChange={[Function]}
        type="text"
        value="climate"
      />
    </form>
  </div>
</SearchBar>
`;
